### PR TITLE
Use the RHEL 9 handler for kickstart parsing (#1961087)

### PR DESCRIPTION
**Pull request by M4rtinK**
Anaconda needs to use the RHEL 9 handler when parsing kickstart on
RHEL 9. Otherwise RHEL only commands & RHEL specific command behavior
will not work.

Also bump minimum Pykickstart version in the spec file to make sure
the handler is available.

Resolves: rhbz#1961087

### CODE REVIEW

**Comment by M4rtinK** [comment]
NOTE: This should be merged only after https://github.com/pykickstart/pykickstart/pull/381 is merged and new Pykickstart version with that PR is available in RHEL 9 repos.

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

**Review by poncovka — APPROVED** [review]

**Review by jkonecny12 — APPROVED** [review]
LGTM!

**Comment by M4rtinK** [comment]
Extended with overrides for commands with RHEL9 versions & BTRFS related commands so that BTRFS is kept deprecated.

Also thanks a lot to @rvykydal for digging up his RHEL8 PRs for this. :)

**Comment by rvykydal** [comment]
I think the failing unit-test is right, RHEL9_BTRFSData is missing in pykickstart (https://github.com/pykickstart/pykickstart/pull/385)

**Review by poncovka — CHANGES_REQUESTED** [review]
Could you please test it?

File: pyanaconda/modules/storage/kickstart.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ class BTRFS(COMMANDS.BTRFS):
 
     def parse(self, args):
         """Parse the command."""
+        # pylint: disable=assignment-from-no-return
```
Comment (poncovka) [inline pyanaconda/modules/storage/kickstart.py:75]: The upstream should be fixed with https://github.com/pykickstart/pykickstart/pull/386.

Comment (M4rtinK) [reply pyanaconda/modules/storage/kickstart.py:75]: Thanks! :)

**Comment by M4rtinK** [comment]
> Could you please test it?

Re-tested with kickstart containing the `syspurpose` command as well as kickstarts checking if btrfs is properly deprecated provided by Radek (thanks! :) ). All seems to work fine.

**Review by poncovka — APPROVED** [review]
Thanks!

**Review by rvykydal — APPROVED** [review]
Looks good to me, thank you.

**Pull request merged**